nncf/torch/sparsity/magnitude/algo.py

```diff
@@ -158,7 +158,7 @@ def compression_stage(self) -> CompressionStage:
         if self._mode == 'local':
             return CompressionStage.FULLY_COMPRESSED
 
-        if self.scheduler.current_sparsity_level == 0:
+        if self.scheduler.current_sparsity_level is None or self.scheduler.current_sparsity_level == 0:
```

Comment:
It is a very difficult criterion for UNCOMPRESSED compression stage. I would recommend return None or create separate method for it.